app/src/timelinecells.cpp

```diff
@@ -444,12 +444,19 @@ void TimeLineCells::paintFrames(QPainter& painter, QColor trackCol, const Layer*
 
     int recHeight = height - 4;
 
-    layer->foreachKeyFrame([&](KeyFrame* key)
+    const QList<int> selectedFrames = layer->getSelectedFramesByPos();
+    for (auto pair : layer->keyframes())
```

Comment:
Doesn't the original `layer->foreachKeyFrame()` fit your need here?

Reply:
Ah.. yes that's right.. I forgot about that.
I've fixed it now.